### PR TITLE
Feature/multi vlan firewall

### PR DESCRIPTION
**Pull request by sumit-tembe-ibm**

### CODE REVIEW

File: ibm/resource_ibm_multi_vlan_firewall.go [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+		return fmt.Errorf("Error during Place order for Creating: %s", err)
+	}
+	_, vlan, err := findDedicatedFirewallByOrderId(sess, *receipt.OrderId, d)
+	if err != nil {
+		return fmt.Errorf("Error during creation of dedicated hardware firewall: %s", err)
+	}
+	id := *vlan.NetworkFirewall.Id
+	d.SetId(fmt.Sprintf("%d", id))
+	d.Set("datacenter", datacentername)
+	d.Set("type", firewalltype)
```
Comment (sakshiag) [inline ibm/resource_ibm_multi_vlan_firewall.go]: Please Set datacenter and firewall_type in read function

File: ibm/resource_ibm_multi_vlan_firewall_test.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+		},
+	})
+}
+
+const testAccCheckIBMMultiVlanFirewallConfig_basic = `
+resource "ibm_multi_vlan_firewall" "firewall_first" {
+	datacenter = "dal13"
+	pod = "pod01"
+	name = "Checkdelete1"
+	public_vlan_id = 2213543
```
Comment (sakshiag) [inline ibm/resource_ibm_multi_vlan_firewall_test.go]: public_vlan_id isa computed field. user cannot provide computed field.

**Comment by sakshiag** [comment]
Please follow the golang naming conventions. 
1) Use camelcase for all the variables .

Please remove the hardcoded error messages and return the api error.

File: ibm/resource_ibm_multi_vlan_firewall.go [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+			"private_vlan_id": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"firewall_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAllowedStringValue([]string{"FortiGate Firewall Appliance HA Option", "FortiGate Security Appliance"}),
```
[the remark this answers is not in the document]
Comment (sumit-tembe-ibm) [reply ibm/resource_ibm_multi_vlan_firewall.go]: It is already in validators.go

[the remark this answers is not in the document]
Comment (sakshiag) [reply ibm/resource_ibm_multi_vlan_firewall.go]: Yeah it is defined... the values which you are passing can be defined as a map in the validators.go. you can refer other methods for eg. https://github.com/IBM-Cloud/terraform-provider-ibm/blob/4a2906bf70070de2fda772d4a4ea2ce0ea372007/ibm/validators.go#L475

File: ibm/resource_ibm_multi_vlan_firewall.go [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+			"password": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"addon_configuration": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				ForceNew:    true,
```
[the remark this answers is not in the document]
Comment (sumit-tembe-ibm) [reply ibm/resource_ibm_multi_vlan_firewall.go]: Not able to find the method to update addon_configuration through terraform. Looks like terraform doesn't support this but Softlayer portal does.

[the remark this answers is not in the document]
Comment (hkantare) [reply ibm/resource_ibm_multi_vlan_firewall.go]: https://github.com/softlayer/softlayer-go/blob/3965b0d2a64bc98580f52e32867fdbad47cbe3e9/datatypes/container.go#L4119
Please use this to update addon_configuration , Get the list of addons get the prices and add to the order.

File: ibm/resource_ibm_multi_vlan_firewall.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (sumit-tembe-ibm) [reply ibm/resource_ibm_multi_vlan_firewall.go]: What we have written is the same way how it is consumed in other resources. Not sure what to change here? The line which you had commented on is the line from where the function is getting called and we are using existing function for it.

**Pull request merged**